### PR TITLE
WB-1732: Add semantic colors as tokens

### PR DESCRIPTION
**Pull request by jandrade**
## Summary:

Added a new `semanticColor` export in `wonder-blocks-tokens` that includes all
the semantic colors as tokens. This will allow us to use these colors in our
components and themes.

Added documentation for the new tokens in the foundations section of the
Wonder Blocks documentation, and also a new section in the tokens section of
the `wonder-blocks-tokens` documentation.

Issue: https://khanacademy.atlassian.net/browse/WB-1732

## Test plan:

Navigate to the following pages and verify that the docs are informative and
correct:

- /?path=/docs/foundations-using-color--docs
- /?path=/docs/wonder-blocks-tokens-tokens-semantic-color--docs

Screenshot 2024-08-15 at 5 25 51 PM

Screenshot 2024-08-15 at 5 25 59 PM

### CODE REVIEW

File: __docs__/wonder-blocks-tokens/__overview__.mdx [hunk 1]
```diff
@@ -20,10 +20,11 @@ building blocks of our design system.
...
 
--   <a href="../?path=/docs/packages-tokens-border--docs">Border</a>
--   <a href="../?path=/docs/packages-tokens-color--docs">Color</a>
--   <a href="../?path=/docs/packages-tokens-spacing--docs">Spacing</a>
--   <a href="../?path=/docs/packages-tokens-typography--docs">Typography</a>
+-   <a href="/?path=/docs/packages-tokens-border--docs">Border</a>
+-   <a href="/?path=/docs/packages-tokens-semantic-color--docs">Semantic Colors</a>
+-   <a href="/?path=/docs/packages-tokens-color--docs">Color Primitives</a>
+-   <a href="/?path=/docs/packages-tokens-spacing--docs">Spacing</a>
+-   <a href="/?path=/docs/packages-tokens-typography--docs">Typography</a>
```
Comment (jandrade) [inline __docs__/wonder-blocks-tokens/__overview__.mdx:27]: **note:** Reworked this section to include semantic tokens + differentiate them from the primitive ones.

/?path=/docs/packages-tokens-overview--docs

Screenshot 2024-08-15 at 5 44 17 PM

File: __docs__/wonder-blocks-tokens/tokens-color.mdx [hunk 2]
```diff
@@ -14,7 +15,30 @@ import packageConfig from "../../packages/wonder-blocks-tokens/package.json";
...
+                        href="/?path=/docs/packages-tokens-semantic-color--docs"
+                        target="_blank"
+                    >
+                        Semantic Color tokens
+                    </a>
+                ),
+            },
+        ]}
+    />
+</View>
```
Comment (jandrade) [inline __docs__/wonder-blocks-tokens/tokens-color.mdx:41]: **note:** Added a banner to state that using semantic tokens is now our recommended approach for new stuff.

Screenshot 2024-08-15 at 5 45 11 PM

File: packages/wonder-blocks-tokens/src/tokens/color.ts [hunk 3]
```diff
@@ -79,14 +81,18 @@ export const color = {
...
     fadedGreen16: fadedColorWithWhite(baseColors.green, 0.16),
+    fadedGreen8: fadedColorWithWhite(baseColors.green, 0.08),
     // Gold shades
     fadedGold24: fadedColorWithWhite(baseColors.gold, 0.24),
     fadedGold16: fadedColorWithWhite(baseColors.gold, 0.16),
+    fadedGold8: fadedColorWithWhite(baseColors.gold, 0.08),
     // Purple shades
     fadedPurple24: fadedColorWithWhite(baseColors.purple, 0.24),
     fadedPurple16: fadedColorWithWhite(baseColors.purple, 0.16),
+    fadedPurple8: fadedColorWithWhite(baseColors.purple, 0.08),
```
Comment (jandrade) [inline packages/wonder-blocks-tokens/src/tokens/color.ts:92]: **note:** New color as we agreed with design. https://www.figma.com/design/pmQkmkD4FwfIjEPVVtFn9Q?node-id=16516-1062#847262137

File: packages/wonder-blocks-tokens/src/tokens/color.ts [hunk 4]
```diff
@@ -79,14 +81,18 @@ export const color = {
...
     // Khanmigo
-    eggplant: "#5f1e5c",
+    eggplant: eggplant,
+    fadedEggplant8: fadedColorWithWhite(eggplant, 0.08),
```
Comment (jandrade) [inline packages/wonder-blocks-tokens/src/tokens/color.ts:95]: **note:** New color requested from Design.

File: packages/wonder-blocks-tokens/src/tokens/semantic-color.ts
Comment (jandrade) [inline packages/wonder-blocks-tokens/src/tokens/semantic-color.ts:1]: **note:** These are the main implementation changes in this PR.

**Comment by jandrade** [comment]
@beaesguerra 

> And maybe we can add headings in between each group for the semantic tokens, what do you think?

This would be already covered in the "Using Color" page. I kept everything under the same table to be consistent with the rest of the existing tokens.

I like the idea of grouping Primitives in the same way we do it in Figma... I'll work on that and post the result here.

File: __docs__/foundations-color.mdx [hunk 5]
```diff
@@ -1,246 +1,174 @@
...
     />
-</ColorPalette>
+</View>
 
-## Categories
+### Status
 
-### Product
+For labels, icons, filters, alerts, and other elements where color can add
+meaning to the state of the system or an item in the system.
```
[the remark this answers is not in the document]
Comment (jandrade) [reply __docs__/foundations-color.mdx:96]: @beaesguerra Thanks for the suggestion! I added a success Banner to show that info:

https://5e1bf4b385e3fb0020b7073c-mfciubzoyd.chromatic.com/?path=/docs/foundations-using-color--docs#status

Screenshot 2024-08-21 at 11 21 55 AM

I added a similar one in the `Text` section and also added an `Accessibility` section at the end of the page, which includes more links to WCAG.

https://5e1bf4b385e3fb0020b7073c-mfciubzoyd.chromatic.com/?path=/docs/foundations-using-color--docs#accessibility

**Review by beaesguerra — APPROVED** [review]
Looks great!! 🚀  Thanks for adding in those updates 😄

**Pull request merged**